cmd/clusterctl/phases/pivot.go

```diff
@@ -253,8 +253,12 @@ func moveSecret(from sourceClient, to targetClient, secret *corev1.Secret) error
 	// New objects cannot have a specified resource version. Clear it out.
 	secret.SetResourceVersion("")
 
-	// Remove owner reference.
-	secret.SetOwnerReferences(nil)
+	// Remove owner reference only if they're part of our known APIGroup
+	for _, ownerRef := range secret.OwnerReferences {
```

Comment:
If we remove the ownerReferences here, then the Secrets will be orphaned if the `Cluster` is deleted post pivot, since the Secrets are not adopted by the Cluster.

I'm wondering if we should also keep track of which secrets had ownerReferences, and set the proper ownerReferences after the `Cluster` has been migrated? @vincepri thoughts?

Reply:
Maybe we can do that instead, re-setting the owner references after being created, because I don't think they'll be actually reconciled after they're created. @tahsinrahman Do you think you could tackle that as part of the PR?

Reply:
sure, i'll try!